File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        countr = Counter(nums)
        sortr = [[] for i in range(len(nums) + 1)]
        res = []

        for val, count in countr.items():
            sortr[count].append(val)

        for i in range(len(nums),0,-1):
            for n in sortr[i]:
                res.append(n)
                if len(res) == k:
                    return res

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        //Create map to store values
        unordered_map<int, int> valueToFrequency;
        multimap<int,int> frequencyToValue;
        vector<int> solution;

        for(int num: nums){
            valueToFrequency[num] += 1;
        }

        for(auto pair : valueToFrequency){
            frequencyToValue.insert({pair.second,pair.first});
        }

        //iterator to end of container
        auto iter = frequencyToValue.end();
        iter--;

        while(k > 0){
            solution.push_back(iter->second);
            iter--;
            k--;
        }

        return solution;
    }
};